gc.ts

```diff
@@ -380,20 +381,63 @@ export class MnS<A extends MarkableAllocator> {
         }
 
         case TAG_LIST: {
-          throw new Error("TODO: trace list");
-        }
+          // Layout: [32-bit TAG_LIST, 32-bit <length>, 32-bit <capacity>, data...]
+
+          // Extract value at childPtr + 4. Assumed to be a primitive value
+          const listLength = childSize;
+          
+          // Sanity check, just-in-case
+          // NOTE(sagar): probably not necessary
+          if(isPointer(listLength)) {
+            throw new Error("Pointer value stored in the place of list length");
+          }
```

Comment:
You'll need to check with the List team or check `compiler.ts`. I'm not sure if the length is a tagged value or not

Reply:
If it's tagged, you'll need to shift it 1 bit right.

Reply:
@student-of It's an untagged value